hypershift-operator/controllers/scheduler/autoscaler.go

```diff
@@ -206,10 +206,12 @@ func nodeMachineSetsToScaleDown(node *corev1.Node, machineSets []machinev1beta1.
 	pairLabel := node.Labels[OSDFleetManagerPairedNodesLabel]
 	if pairLabel != "" {
 		nodesToScaleDown = filterNodes(nodes, func(n *corev1.Node) bool {
-			return n.Labels[OSDFleetManagerPairedNodesLabel] == pairLabel
+			return n.Labels[OSDFleetManagerPairedNodesLabel] == pairLabel && n.Labels[hyperv1.NodeSizeLabel] != ""
```

Comment:
shouldn't we instead return early in the MachineSetDescaler.Reconcile if node.Labels[hyperv1.NodeSizeLabel] == ""?